tools/CompileReleaseNotes.py

```diff
@@ -0,0 +1,192 @@
+
+
+if __name__ == "__main__":
+    # The release notes always refer to the repository that contains this file
+    repo = git.Repo(os.path.dirname(os.path.dirname(__file__)))
+
+    import argparse
+    parser = argparse.ArgumentParser(
+        description=("Compile release notes based on merged pull-requests. "
+                     "Repository: {}. Branch: {}.").format(
+                         repo.working_dir, repo.active_branch),
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '--output',
+        '-o',
+        required=False,
+        help=("Name of the output file, e.g. 'release_notes.md'."))
+    parser.add_argument(
+        '--from',
+        required=False,
+        dest='from_rev',
+        help=(
+            "Commit ID (or tag) that marks the last release. Defaults to the "
+            "most recent tag with format YYYY-MM-DD."))
+    parser.add_argument('--to',
+                        required=False,
+                        dest='to_rev',
+                        default='HEAD',
+                        help=("Commit ID (or tag) that marks this release."))
+    parser.add_argument(
+        '--github-repository',
+        required=False,
+        default='sxs-collaboration/spectre',
+        help=("GitHub repository associated with pull-request IDs in merge "
+              "commits."))
+    parser.add_argument(
+        '--github-token',
+        required=False,
+        help=
+        ("Access token for GitHub queries. Refer to the GitHub documentation "
+         "for instructions on creating a personal access token."))
+    parser.add_argument('-v',
+                        '--verbose',
+                        action='count',
+                        default=0,
+                        help="Verbosity (-v, -vv, ...)")
+    parser.add_argument('--silent',
+                        action='store_true',
+                        help="Disable any logging")
+    args = parser.parse_args()
+
+    # Set the log level
+    logging.basicConfig(
+        level=logging.CRITICAL if args.silent else (logging.WARNING -
+                                                    args.verbose * 10))
+
+    # Retrieve last release
+    if args.from_rev is None:
+        args.from_rev = get_last_release(repo=repo, head_rev=args.to_rev)
+        logging.info("Last release is: {}".format(args.from_rev))
+
+    # Retrieve list of merged PRs for this release
+    merged_prs = get_merged_pull_requests(repo=repo,
+                                          from_rev=args.from_rev,
+                                          to_rev=args.to_rev)
+    logger.info("Merged PRs since last release:\n{}".format('\n'.join(
+        map(str, merged_prs))))
+
+    # Try to query GitHub for further information on the merged PRs
+    breaking_prs = []
+    feature_prs = []
+    bugfix_prs = []
+    if args.github_token:
+        import github
+        import tqdm
+        gh = github.Github(args.github_token)
```

Comment:
I find it surprising you need a token to retrieve public repo info. Is this just GitHub trying to limit web crawlers or something like that?

Reply:
Yes. But as it turns out GitHub also allows [60 unauthenticated requests per hour](https://docs.github.com/en/free-pro-team@latest/rest/overview/resources-in-the-rest-api#rate-limiting) based on the IP address, so I'll change the script to allow that.

Reply:
Just to make sure we're on the same page: you're saying you'll make it so you can either specify an access token _or_ hope you haven't hit the 60 per hour limit?

Reply:
Yes. And add an opt-out CLI option.

Reply:
Okay, great! Thank you!